### ISSUE TITLE
[P1] Fail closed instead of fitting TF-IDF likelihoods on held-out evaluation data

### ISSUE THREAD

**Issue by ankaggarwal94** [ankaggarwal94/qanta-buzzer#35]
## Summary

The split-aware pipeline can silently stop being held out when the training/reference artifact is missing, empty, corrupt, or no longer colocated with the evaluation artifact.

At current `main` (`5cafae3c4fc3dfd1525975aa34fda77975b7f4cb`):

- `scripts/run_baselines.py` prefers the training corpus for likelihood construction, but falls back to the selected validation/evaluation questions when train loading fails.
- `scripts/evaluate_all.py` does the same for the softmax-profile evaluation.
- PPO replay first tries the recorded reference path, then a sibling train split, and finally the evaluation split.
- `scripts/_common.py::build_likelihood_model()` fits TF-IDF vocabulary and IDF weights on every question text and option profile supplied to it.

The fallback therefore uses held-out text and held-out option profiles to fit the representation used to score those same held-out examples. A console warning is not enough: the command still exits successfully and emits a normal-looking `evaluation_report.json`, while the report does not record the actual likelihood-reference path/split/hash for the softmax result.

## Relevant code

- [`scripts/run_baselines.py`](https://github.com/ankaggarwal94/qanta-buzzer/blob/5cafae3c4fc3dfd1525975aa34fda77975b7f4cb/scripts/run_baselines.py)
- [`scripts/evaluate_all.py`](https://github.com/ankaggarwal94/qanta-buzzer/blob/5cafae3c4fc3dfd1525975aa34fda77975b7f4cb/scripts/evaluate_all.py)
  - `likelihood_questions = mc_questions` when train is unavailable.
  - PPO replay also falls back to `mc_questions`.
- [`scripts/_common.py`](https://github.com/ankaggarwal94/qanta-buzzer/blob/5cafae3c4fc3dfd1525975aa34fda77975b7f4cb/scripts/_common.py)
  - `build_likelihood_model()` creates the TF-IDF corpus from `q.question` and every `option_profile` in the supplied rows.

## Minimal reproduction

1. Build normal split artifacts.
2. Remove or corrupt `train_dataset.json` while retaining `test_dataset.json`.
3. Run:

```bash
python scripts/evaluate_all.py \
  --mc-path artifacts/main/test_dataset.json \
  likelihood.model=tfidf
```

### Actual behavior

The script prints a warning equivalent to:

```text
train split missing or empty; using eval split
```

then fits TF-IDF on the test questions/profiles and writes the standard report.

### Expected behavior

For any train-dependent likelihood model, held-out evaluation should fail closed unless the operator explicitly selects an in-sample diagnostic mode. A missing/corrupt reference corpus is a broken evaluation contract, not a recoverable convenience fallback.

## Impact

- Test-set vocabulary and IDF statistics influence the representation used to score the test set.
- Test option profiles become part of the fitted corpus.
- A moved artifact directory or pruned training split can turn a formerly held-out run into an in-sample run without changing the command's exit status.
- Report consumers cannot reliably tell whether the softmax result was train-fit or test-fit.
- PPO replay can use a different likelihood corpus from training, changing its observations as well as contaminating evaluation.

The severity is highest for TF-IDF. Other likelihood families should still record and enforce their reference-corpus contract because profile construction, caches, or future trainable adapters may depend on it.

## Proposed resolution

- Default behavior: raise a clear error when a required training/reference corpus is absent, empty, unreadable, or does not match recorded provenance.
- Add an explicit diagnostic override such as `--allow-in-sample-likelihood`; when used, mark the artifact as contaminated/non-held-out and make downstream audit-card consumers refuse a confirmatory verdict.
- Persist a reference-corpus binding for every evaluated agent:
  - resolved path;
  - logical split;
  - row count;
  - ordered QID/content hash;
  - profile/config hash;
  - whether fallback was used.
- For PPO checkpoints, verify the recorded corpus hash rather than silently substituting a sibling file with the same name.

## Acceptance criteria

- [ ] Missing, corrupt, or empty train/reference data fails by default in `run_baselines.py`, `evaluate_all.py`, and PPO replay.
- [ ] An explicit in-sample override is required to continue and produces a non-confirmatory/WARN artifact status.
- [ ] `evaluation_report.json` records the actual likelihood-reference split/path/hash/count for softmax and PPO independently.
- [ ] PPO replay refuses a corpus whose content hash differs from the training record, even if the path/name matches.
- [ ] Regression tests cover missing, empty, malformed, moved, and content-modified train artifacts.
- [ ] A false-oracle test proves that test-only vocabulary cannot enter the train-fit TF-IDF vocabulary.
- [ ] Existing legacy combined-data behavior remains available only behind an explicit opt-in and is labeled `in_sample` rather than `test`.

## Validation plan

Use a toy train corpus containing only token `TRAIN_ONLY` and a test corpus containing only `TEST_ONLY`. Under the held-out path, the fitted TF-IDF vocabulary must not contain `TEST_ONLY`. Deleting the train artifact must fail; enabling the explicit diagnostic override may proceed but must stamp the output as in-sample and non-confirmatory.

**Comment by ankaggarwal94** [comment]
## Independent CoVe revalidation

**Verdict:** CONFIRMED at `main` `5cafae3c4fc3dfd1525975aa34fda77975b7f4cb`  
**Severity:** P1 retained  
**Confidence:** high

Independent source traces agree on the failure chain:

1. `run_baselines.py` and the softmax path in `evaluate_all.py` substitute the selected eval rows when the sibling train/reference corpus is absent, empty, or unreadable.
2. PPO replay falls through from the recorded reference path to a sibling train file and then to the evaluation rows.
3. `_common.build_likelihood_model()` constructs the TF-IDF corpus from every supplied question text and option profile, and `TfIdfLikelihood.fit()` learns vocabulary/IDF from that corpus.
4. The emitted report records the logical evaluation split, but not a content binding proving which reference corpus actually fit the softmax or replay likelihood.

**Disconfirmation/mitigation found:** the normal split-aware happy path does use the train corpus, and the fallback prints a console warning. Therefore this issue does not claim all current artifacts are contaminated, nor that the fallback is literally invisible to an attentive operator. The defect is that the command succeeds and emits a normal success-schema artifact without encoding the contamination in that artifact.

No matching open or closed issue was found. Issue #23 concerns MC construction with an explicitly empty reference list and is not a duplicate of evaluation-time held-out contamination.

A fresh runtime reproduction was not executed in this connector-only audit; the verified control flow is direct and deterministic. The proposed `TRAIN_ONLY` / `TEST_ONLY` false oracle remains the right executable regression.
